eya_def_tools/eya_def_tools/data_models/dataset.py

```diff
@@ -74,14 +160,12 @@ class DatasetStatistic(EyaDefBaseModel):
             "not be empty if the field is included."
         ),
     )
-    statistic_type: BasicStatisticType | ExceedanceLevelStatisticType = pdt.Field(
+    statistic: AnyStatisticType = pdt.Field(
```

Comment:
The substantial change in this PR is to refine and extend how a statistic is defined. Now any statistic can have a time frame start date, time frame end date and return period, where relevant. The probability of exceedance levels also have the probability level, as before.